ckanext/s3filestore/tests/test_uploader.py

```diff
@@ -190,23 +190,24 @@ def _test_dataset(self, private=False, title='Test Dataset', author='test'):
             author=author,
             owner_org=self.organisation['id'])
 
-    def _upload_test_resource(self, dataset=None):
+    def _upload_test_resource(self, dataset=None, filename='data.csv'):
         ''' Creates a test resource in the specified dataset
         by uploading a file.
         '''
         if not dataset:
             dataset = self._test_dataset()
-        file_path = os.path.join(os.path.dirname(__file__), 'data.csv')
+        file_path = os.path.join(os.path.dirname(__file__), filename)
         return helpers.call_action(
             'resource_create',
             package_id=dataset['id'],
             upload=FlaskFileStorage(io.open(file_path, 'rb')),
-            url='data.csv')
+            url=filename)
 
     def test_resource_upload(self):
```

Comment:
please add a confusion test, where someone tries to upload a pdf that is actually a csv, and a csv as a pdf.

I feel we need to ensure we are file safe for pdf, csv, xls/xlsx as these get opened automatically in the browser (pdf) and xloader/validation plugin tries to read said data and we don't want to allow low probability ddos or other side attacks (or other events on other sites getting files form us) in allowing non valid file's into our storage.

Reply:
confusion test would still be good, that file type ending is used.
and if ending is missed, it goes through the mime magic and does it best (i.e if for some reason file type is stripped or mangled)

Reply:
Those tests exist in `ckanext-resource-type-validation`, which attempts to cover all the bases for mismatched files and types.

Reply:
2048 bytes, see https://github.com/qld-gov-au/ckanext-resource-type-validation/blob/42e28898be66f44b96f4d366725968a9f7195cd2/ckanext/resource_type_validation/resource_type_validation.py#L95